controllers/mongodb_tls.go

```diff
@@ -37,11 +39,20 @@ func (r *ReplicaSetReconciler) validateTLSConfig(mdb mdbv1.MongoDBCommunity) (bo
 
 	r.log.Info("Ensuring TLS is correctly configured")
 
-	// Ensure CA ConfigMap exists
-	caData, err := configmap.ReadData(r.client, mdb.TLSConfigMapNamespacedName())
+	// Ensure CA cert is configured
+	var caResourceName types.NamespacedName
+	var caResourceType string
+	if mdb.Spec.Security.TLS.CaCertificateSecret != nil {
+		caResourceName = mdb.TLSCaCertificateSecretNamespacedName()
+		caResourceType = "Secret"
+	} else {
+		caResourceName = mdb.TLSConfigMapNamespacedName()
+		caResourceType = "ConfigMap"
+	}
+	caData, err := configmap.ReadData(r.client, caResourceName)
```

Comment:
[q] how does this work? I understand if the CA is passed through config map, but if it's a secret how can we read it? I expect it to fail?

Reply:
thanks for spotting this. It was working probably because the secret and the config map have a similar struct signature but I made the change so that we're explicitly reading a secret or a configmap